apps/passport-client/src/appHooks.ts

```diff
@@ -168,3 +169,32 @@ export function useRequirePassword() {
     });
   }
 }
+
+// Hook that enables keystrokes to properly listen to laser scanning inputs from supported devices
+export function useLaserScannerKeystrokeInput() {
+  const [typedText, setTypedText] = useState("");
+
+  useEffect(() => {
+    function handleKeyDown(event: KeyboardEvent) {
+      if (event.key === "Enter") {
+        // Check URL regex and navigate to the last match, if it exists
+        const url = getLastValidURL(typedText);
+        if (url) {
+          window.location.href = url;
```

Comment:
now that i see what this is doing i'm pretty sure we don't want this on any other screen than the scan screen.

Reply:
remove on all except ScanScreen